sample/todo-sync/src/com/cloudant/todo/TasksModel.java

```diff
@@ -233,8 +231,8 @@ public void reloadReplicationSettings()
         // Set up the new replicator objects
         URI uri = this.createServerURI();
 
-        mPullReplicator = ReplicatorBuilder.pull().to(mDatastore).from(uri).build();
-        mPushReplicator = ReplicatorBuilder.push().from(mDatastore).to(uri).build();
+        mPullReplicator = ReplicatorBuilder.pull().to(mDocumentStore.database).from(uri).build();
+        mPushReplicator = ReplicatorBuilder.push().from(mDocumentStore.database).to(uri).build();
```

Comment:
Yeah it would, although IIRC we don't update indexes automatically after a replication so we should be careful to not imply that intent with the argument. On the plus side passing the `DocumentStore` gives us scope to improve that.